added dynamic z axis option (#5025)
sunpy/sunpy: changelog/5025.feature.rst

> Added dynamic z axis option to the animators.

Reviewer: This changelog needs expanding.

Added the ability to dynamically scale `sunpy.visualization.animator` instances.

> By specifying ``vmax="auto"`` will now clip the minimum and maximum at each slider step to 1% and 90% respectively.

Reviewer: ```suggestion
By specifying ``vmax="auto"``, it will now clip the minimum and maximum at each slider step to 1% and 90% respectively.
```

> By specifying ``vmax="auto"``, it will now clip the minimum and maximum at each slider step to 1% and 90% respectively.

Reviewer 2: We should make these limits symmetrical, e.g. use 1% and 99% or 10% and 90%. I think there's somewhere else in the library that we use 1% and 99%, so I would say lets go with that.

Author: @dstansby 99% causes the image to not be as visible but if that's the standard then I shall make that change.
I'll make the change for 2 specifying either of them separately.
While writing tests should I write 3 separate tests for ```vmax='auto'``` , ```vmin='auto'```, and both, or should I just have one test where both ```vmax='auto'``` and ```vmin='auto'```?

> By specifying a ``clip_interval``, it will now clip the minimum and maximum at each slider step to the specified interval.

Reviewer: ```suggestion
By specifying the ``clip_interval`` keyword, it will now clip the minimum and maximum at each slider step to the specified interval.
```

> Added dynamic z axis option to `sunpy.visualization.animator`.

Reviewer: Is this a z axis?

Author: Was confused about that actually, when I asked about it in the #4718 this is what @dstansby stated so I was under the assumption that it's correct.

Reviewer: This sets the image min and max limits on a 2D image. I don't think that is a Z axis.

Author: Yeah, so I guess the PR as a whole was pointing to a separate issue being mentioned under #4718

Reviewer: No this does address that issue.

Author: So, should I change the name of this PR to something else?

Reviewer: The name of the PR doesn't really matter.

> Added dynamic z axis option to `sunpy.visualization.animator`.
> Specifying ``vmax='auto'`` calculates and updates the colormap
> according to the current slice shown.

Reviewer: ```suggestion
Added the ability to dynamically scale `sunpy.visualization.animator` instances.
By specifying ``vmax="auto"`` will now clip the minimum and maximum at each slider step to 1% and 90% respectively.
```
